### DISCUSSION TITLE
Can I set a time-varying inlet flow in THM？

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#26153] [Q&A Modules: Thermal Hydraulics] [answered]
@joshuahansel Hello josh, I'm excited that THM finally has its own Q&A module.
To get back on track, I want to calculate a case which has a time-varying inlet mass flow. Is there any way to achieve this？

Ethan

**Comment by joshuahansel** [comment, marked as answer]
Hi Ethan,

Yes, you would use some `ControlLogic` with `InletMassFlowRateTemperature1Phase`. See `modules/thermal_hydraulics/test/tests/components/inlet_mass_flow_rate_1phase/clg.ctrl_m_dot_3eqn_rdg.i`. Let me know if you have any questions.

**Reply by Ethan-xj** [reply]
Thanks a lot, it works. And I have another issue now. I can calculate a area-variation case using `FlowChannel1Phase` in THM. But in the subcahnnel model I created, even if I only change the area in one element of one channel, the calculation is non-convergent, like:
```
Time Step 1, time = 0.25, dt = 0.25
 0 Nonlinear |R| = 2.557412e+00
      0 Linear |R| = 2.557412e+00
      1 Linear |R| = 1.664515e-04
      2 Linear |R| = 9.914895e-11
 1 Nonlinear |R| = 1.451551e+11
Nonlinear solve did not converge due to DIVERGED_DTOL iterations 1
 Solve Did NOT Converge!
Aborting as solve did not converge
```
I am confused. Theoretically, I only added the lateral momentum equation and the lateral term. Do you have any ideas regarding the reasons for non-convergence?

**Reply by joshuahansel** [reply]
Do you know that it's related to your terms? Are you able to test a corresponding case without your terms?

**Reply by Ethan-xj** [reply]
I think I found the problem. I still use UO to transfer the variables of adjacent channels(like rho = rhouA/rhoA). And I use AREA_LINEAR. When I change AREA_LINEAR into AREA, the area-variation case can be converged. Btw, if the area doesn't change, there's no difference between AREA_LINEAR and AREA right?

**Reply by joshuahansel** [reply]
> if the area doesn't change, there's no difference between AREA_LINEAR and AREA right?

Correct. And in most cases, you want to use `AREA`, the cell-average cross-sectional area corresponding to the solution variables themselves.
